rust/lance-linalg/src/kmeans.rs

```diff
@@ -446,37 +447,44 @@ impl KMeans {
         let n = data.len() / self.dimension;
         let metric_type = self.metric_type;
         const CHUNK_SIZE: usize = 1024;
+
+        // Normalized centroids for fast cosine.
```

Comment:
uber nit: clarify this allows us to skip one of the norm calculation in `A.B/(|A||B|)`

Reply:
Added